lib/ctblgrp.gi

```diff
@@ -412,25 +386,28 @@ local i,pcomp,m,r,D,neue,tm,news,opt;
   od;
 
   if opt<>true then
-    pcomp:=NullspaceMat(D.projectionMat*TransposedMat(D.modulars));
-    for i in [1..Length(D.raeume)] do
-      r:=D.raeume[i];
-      if r.dim = Length(r.base[1]) then
-	# trivial case: Intersection with full space in the beginning
-	r:=rec(base:=pcomp);
-      else
-	r:=rec(base:=SumIntersectionMat(pcomp,r.base)[2]);
-      fi;
-      r.dim:=Length(r.base);
-      # note stabilizer
-      if IsBound(D.raeume[i].stabilizer) then
-	r.stabilizer:=D.raeume[i].stabilizer;
-      fi;
-      if r.dim>0 then
-	DxActiveCols(D,r);
-      fi;
-      D.raeume[i]:=r;
-    od;
+    pcomp:=NullspaceMat(D.projectionMat*TransposedMatImmutable(Matrix(D.field,D.modulars)));
+    if Length(pcomp)>0 then
+      for i in [1..Length(D.raeume)] do
+        r:=D.raeume[i];
+        if r.dim = Length(r.base[1]) then
+          # trivial case: Intersection with full space in the beginning
+          r:=rec(base:=pcomp);
+        else
+          r:=rec(base:=SumIntersectionMat(pcomp,
+            Matrix(BaseDomain(r.base[1]),r.base))[2]);
```

Comment:
Would it perhaps make sense to make sure that the `base` component is a `MatrixObj` from the beginning, then calling `Matrix` here would not be necessary?

Reply:
It could be. In the old setup a matrix is just a list of vectors, and there was no need to distinguish the concepts. Here, one has to, and it seemed naively that a basis should be a list of vectors rather than a matrix.